### PR TITLE
DOC: Expand docstrings for head / tail methods

### PR DESCRIPTION
**Pull request by yosukeBaya4**
This is just a first draft of some documentation changes to head and tail methods.

From EuroPython sprint.

### CODE REVIEW

**Review by TomAugspurger — COMMENTED** [review]
Thanks, you can add a `Returns` section and a `See Also` as well if you'd like.

**Comment by TomAugspurger** [comment]
And don't worry about the pycodestyle bot. We're testing it out.

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2855,12 +2855,22 @@ def filter(self, items=None, like=None, regex=None, axis=None):
     def head(self, n=5):
         """
         Returns first n rows
+
+        Parameters
+        ----------
+        n : int, optional
```
Comment (gfyoung) [inline pandas/core/generic.py]: Instead of saying "optional", write "default 5"

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2855,12 +2855,22 @@ def filter(self, items=None, like=None, regex=None, axis=None):
...
         """
         return self.iloc[:n]
 
     def tail(self, n=5):
         """
         Returns last n rows
+
+        Parameters
+        ----------
+        n : int, optional
```
Comment (gfyoung) [inline pandas/core/generic.py]: Same comment as <a href="https://github.com/pandas-dev/pandas/pull/16941#pullrequestreview-50191725">above</a>

**Comment by jreback** [comment]
can you update according to comments

**Comment by jreback** [comment]
pls rebase / update. @gfyoung if you can do this if OP doesn't respond in a while.

**Comment by gfyoung** [comment]
> @gfyoung if you can do this if OP doesn't respond in a while.

Sure thing!

**Comment by gfyoung** [comment]
@jreback : I'm going to assign this to myself.  Feel free to ping if this hasn't been completed before `0.21.0`, and I'll carry it home.

File: pandas/core/generic.py (hunk 2 above)
Comment (gfyoung) [inline pandas/core/generic.py]: Done.

File: pandas/core/generic.py (hunk 1 above)
Comment (gfyoung) [inline pandas/core/generic.py]: Done.

**Comment by gfyoung** [comment]
@jreback : All review comments addressed.  PTAL.

**Comment by jorisvandenbossche** [comment]
Thanks @yosukeBayashi and @gfyoung !

**Pull request merged**